src/main/webapp/app/statistics/quiz-point-statistic/show-quiz-point-statistic.controller.js

```diff
@@ -0,0 +1,240 @@
+
+            // reset old data
+            label = [];
+            backgroundColor = [];
+            ratedData = [];
+            unratedData = [];
+
+            //set data based on the pointCounters
+            vm.quizPointStatistic.pointCounters.forEach(function (pointCounter) {
+                label.push(pointCounter.points);
+                ratedData.push(pointCounter.ratedCounter);
+                unratedData.push(pointCounter.unRatedCounter);
+                backgroundColor.push("#428bca");
+            });
+            // order the bars ascending on points
+            order();
+
+            barChartData.labels = label;
+
+            // load data into the chart
+            // if vm.rated == true  -> load the rated data
+            if (vm.rated) {
+                vm.participants = vm.quizPointStatistic.participantsRated;
+                barChartData.participants = vm.quizPointStatistic.participantsRated;
+                barChartData.datasets.forEach(function (dataset) {
+                    dataset.data = ratedData;
+                    dataset.backgroundColor = backgroundColor;
+                });
+            }
+            //else: load the unrated data
+            else {
+                vm.participants = vm.quizPointStatistic.participantsUnrated;
+                barChartData.participants = vm.quizPointStatistic.participantsUnrated;
+                barChartData.datasets.forEach(function (dataset) {
+                    dataset.data = unratedData;
+                    dataset.backgroundColor = backgroundColor;
+                });
+            }
+            window.myChart.update();
+
+        }
+
+        // switch between the rated and the unrated Results
+        function switchRated() {
+            if(vm.rated) {
+                //load unrated Data
+                barChartData.datasets.forEach(function (dataset) {
+                    dataset.data = unratedData;
+                });
+                vm.participants = vm.quizPointStatistic.participantsUnrated;
+                barChartData.participants = vm.quizPointStatistic.participantsUnrated;
+                vm.rated = false;
+            }
+            else{
+                //load rated Data
+                barChartData.datasets.forEach(function (dataset) {
+                    dataset.data = ratedData;
+                });
+                vm.participants = vm.quizPointStatistic.participantsRated;
+                barChartData.participants = vm.quizPointStatistic.participantsRated;
+                vm.rated = true;
+            }
+            window.myChart.update();
+        }
+
+        //order the data, so that they are ascending
+        function order() {
+            var old = [];
+            while (old.toString() !== label.toString()) {
+                old = label.slice();
+                for(var i = 0; i < label.length-1; i ++) {
+                    if(label[i] > label[i+1]) {
+                        var temp = label[i];
```

Comment:
Please add some comments to this function to make it easier to read and understand.